core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java

```diff
@@ -294,13 +300,17 @@ private void addHookStepToTestCaseMap(Map<String, Object> currentStepOrHookMap,
         else
             hookName = "before";
 
+
         Map<String, Object> mapToAddTo;
         if (hookType == HookType.After || hookType == HookType.Before) {
             mapToAddTo = currentTestCaseMap;
         }
         else {
-            //get latest added step
-            mapToAddTo = currentStepsList.get(currentStepsList.size() - 1);
+            if (hookType == HookType.AfterStep)
```

Comment:
I think it'd be better to explicitly check the hook type.  `if (hookType == HookType.AfterStep || hookType == HookType.BeforeStep) ` and flatten the nested if statement into `else if`.

Reply:
Or a `switch` statement.